src/components/Header/HeaderControls.tsx

```diff
@@ -0,0 +1,118 @@
+/*
+MIT License
+
+Copyright (c) 2021-present, Elastic NV
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton,
+  EuiHealth,
+  useEuiTheme,
+} from "@elastic/eui";
+import React, { useContext } from "react";
+import { Setter } from "../../common/types";
+import { RecordingContext } from "../../contexts/RecordingContext";
+import { TestContext } from "../../contexts/TestContext";
+import { ControlButton } from "../ControlButton";
+import { SaveCodeButton } from "../ExportScriptButton";
+import { TestButton } from "../TestButton";
+
+interface IRecordingHealth {
+  isPaused: boolean;
+  isRecording: boolean;
+}
+
+function RecordingHealth({ isPaused, isRecording }: IRecordingHealth) {
+  if (isRecording && !isPaused)
+    return <EuiHealth color="danger">Recording</EuiHealth>;
+  if (isRecording && isPaused)
+    return <EuiHealth color="warning">Recording paused</EuiHealth>;
+  return <EuiHealth color="subdued">Not recording</EuiHealth>;
+}
+
+interface IHeaderControls {
+  hasActions: boolean;
+  setIsCodeFlyoutVisible: Setter<boolean>;
+}
+
+export function HeaderControls({
+  hasActions: disabled,
+  setIsCodeFlyoutVisible,
+}: IHeaderControls) {
+  const { euiTheme } = useEuiTheme();
+  const { isPaused, isRecording, toggleRecording } =
+    useContext(RecordingContext);
+  const { onTest } = useContext(TestContext);
+  return (
+    <EuiFlexGroup
+      alignItems="center"
+      gutterSize="m"
+      style={{
+        backgroundColor: euiTheme.colors.lightestShade,
+        borderBottom: euiTheme.border.thin,
+        margin: "0px 0px 4px 0px",
+        padding: 8,
+      }}
```

Comment:
Should we encapsulate this into a `styles` map and refer to it within the component instead of inlining? Nbd though, just asking what we usually do in these cases.

Reply:
We don't have a formal decision made around this. Downstream I introduced `styled-components` and am trying to eliminate the inline styles. I made these changes before introducing that. Perhaps we could do a quick sync about how we want to handle custom styles in the project, as it's not something we've devoted attention to in the past.

Reply:
Agreed, as in the previous item, how do you feel about having an issue for this or adding it to a Google Doc which we quickly review weekly as an "Agenda" for the recorder's roadmap?

Reply:
I'm find if we create an issue for this. We can link it in a roadmap agenda then. It won't be sustainable to use inline styles ad hoc, so we should formalize how we want to do this.